### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by Mikey1-nuel**
## OOP school library: add basic UI

**In this project I:**

- Created a `app.rb` file that will serve as your console app entry-point. It should have methods that do the following:
  - List all books.
  - List all people.
  - Created a person (teacher or student, not a plain `Person`).
  - Created a book.
  - Created a rental.
  - List all rentals for a given person id.
- In your `main.rb` define the entry point, this will be a method called `main` that is invoked at the end of your file. This method should do the following:
  - Present the user with a list of options to perform.
  - Lets users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.

### CODE REVIEW

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Mikey1-nuel, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good UI ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- There are some little issues to be fixed for the correct functionality of your app.

- [ ] **In person.rb** Please 🙏🏽 remove false from the method `can_use_services?` otherwise it will return false always 👇🏽 

``` ruby
 def can_use_services?
    of_age? || @parent_permission
  end 
```

- [ ] **In class student and class teacher** Add the argument name to the super to avoid the issue `unknown keyword: :name (ArgumentError)` to be able to create student and teacher.

``` ruby
def initialize(age, parent_permission: true, name: 'Unknown')
    super(age, name, parent_permission: parent_permission)
  end
```

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — COMMENTED** [review]
Hi @Mikey1-nuel, 

# Invalid Code Review Request ❌ 

descarga

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter errors ❌ 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  when 1
+    app.create_student(age, name, parent_permission)
+  when 2
+    puts 'specialization:-'
+    specialization = gets.chomp
+    app.create_teacher(age, specialization, name, parent_permission)
+  else
+    puts 'Invalid input ):'
+  end
+end
```
Comment (abdulhamiid) [inline main.rb]: Example video
image

Your app console
image

According to the project requirements, our console app should behave in the same way as it is in the example. 🤗

Please implement the following to make your app behave in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ)

- 👀 `Has parent permission?` is not required to create a `Teacher`. 🤓 please remove it 👍
- Kindly add a success message that will be displayed when a new item is created. e.g `Person created successfully`

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number(not ID)'
+    @peoples.each.with_index do |person, idx|
+      puts "#{idx}) [#{person.class.name}] Name #{person.name}, ID #{person.id}, Age #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    puts 'Date:-'
+    date = gets.chomp
+    Rental.new(@books[book_index], @peoples[person_index], date)
+  end
```
Comment (abdulhamiid) [inline app.rb]: Your app console
image

Currently, the method `def create_rental` is not working, and creating a rental crash in the app, please kindly fix this issue to ensure that the console app works as expected.. 💪☺

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @Mikey1-nuel 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, I really appreciate the effort you put in! but there are still some which haven't been addressed yet.

almost-there

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Linters checks are passing
- ✔️ Build interactive console apps.
- ✔️ Run a program using the command line.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    case option_id
+    when '4' then create_book
+    when '5' then create_rental
+    when '6' then rental_list
+    when '7' then exit_app
+    else puts 'Choose an existing option!'
+         return
+    end
+    display_options unless option_id == '7'
+  end
```
Comment (KanzaTahreem) [inline app.rb]: - [ ] Well done so far 🥇 The message "Choose an existing option!" is a part of the error handling in the `run_app` and `run_app2` methods. In the `display_options` method, there are two separate calls to `run_app` and `run_app2`, which can lead to the "Choose an existing option!" message appearing even when a valid option is selected ⚠ This was not the case in the last review. Kindly fix this to make sure that your app behave in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) according to the project requirements as mentioned earlier and avoiding any confusion for the user 😃

 image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @Mikey1-nuel 🙋‍♀️

Good job so far! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Good job with the UI
- ✅ Descriptive PR
- ✅ Correct Linters

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕


N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @Mikey1-nuel, 

To highlight 👍🏽 
✔️ App runs as expected
✔️ App matches wireframe

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**